interp/interp_test.go

```diff
@@ -3427,19 +3426,31 @@ var testBuiltinsMap = map[string]func(HandlerContext, []string) error{
 		return os.Link(oldname, newname)
 	},
 	"touch": func(hc HandlerContext, args []string) error {
+		filenames := args // create all arugments as filenames
+
 		newTime := time.Now()
-		if args[0] == "-d" {
-			if !strings.HasPrefix(args[1], "@") {
-				return fmt.Errorf("unimplemented")
+		if args[0] == "-t" {
+			if len(args) < 3 {
+				return fmt.Errorf("usage: touch [-t [[CC]YY]MMDDhhmm[.SS]] file")
+			}
+			filenames = args[2:] // treat the rest of the args as filenames
+
+			arg := args[1]
+			if len(arg) > 15 {
+				return fmt.Errorf("usage: touch [-t [[CC]YY]MMDDhhmm[.SS]] file")
 			}
-			sec, err := strconv.ParseInt(args[1][1:], 10, 64)
+			arg = strings.ReplaceAll(arg, ".", "")
```

Comment:
nit: I think you can just asd the period to the format string passed to time.Parse.